src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -562,6 +566,635 @@ def _encloses_omp_directive(self):
+                # supportable, but need to think
+                if node.operator is \
+                    BinaryOperation.ADD:
+                    index_name = index_name + ("+({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                else:
+                    index_name = index_name + ("-({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                index_strings.append(index_name)
+            elif index_name in firstprivate_list:
+                # name +/- stop-start
+                if node.operator is \
+                    BinaryOperation.ADD:
+                    index_name = index_name + ("+({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                else:
+                    index_name = index_name + ("-({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                index_strings.append(index_name)
+            else:
+                firstprivate_list.append(index_name)
+                # name +/- stop-start 
+                if node.operator is \
+                    BinaryOperation.ADD:
+                    index_name = index_name + ("+({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                else:
+                    index_name = index_name + ("-({0} - {1})"
+                           .format(writer(stop_val)
+                               ,writer(start_val)))
+                index_strings.append(index_name)
+        else:
+            # FIXME I think this should be
+            # supportable, but need to think
+            raise GenerationError(
+                    "Shared variable access used "
+                    "as an index inside an "
+                    "OMPTaskDirective which is not "
+                    "supported.")
+
+    def compute_clauses(self):
+        '''
+        TODO
+        '''
+        from psyclone.psyir.backend.fortran import FortranWriter
+        private_list = []
+        firstprivate_list = []
+        shared_list = []
+        in_list = []
+        out_list = []
+
+        # Create a FortranWriter to convert nodes to strings
+        # TODO: Create issue - support non-Fortran backends
+        writer = FortranWriter()
+
+        # TODO Generate extra clauses
+        # Find the parent Loop node
+        parent_loop = self.ancestor(Loop)
+
+        # Find the parent OMPParallelDirective
+        parallel_directive = self.ancestor(OMPParallelDirective)
+        parallel_private = parallel_directive._get_private_list()
+
+        node = self.children[0].children[0]
+
+        # Find our loop initialisation, variable and bounds
+        loop_var = node.variable
+        start_val = node.children[0]
```

Comment:
Semantic navigation is nicer, e.g. `start_val = node.start_expr`, `stop_expr`, `step_expr`